docs/deployment.rst

```diff
@@ -0,0 +1,145 @@
+.. include:: global.rst.inc
+.. _deployment:
+
+Deployment
+==========
+
+This chapter will give an example how to setup a borg repository server for multiple
+clients.
+
+Machines
+--------
+
+There are multiple machines used in this chapter and will further be named by their
+respective fully qualified domain name (fqdn).
+
+* The backup server: `backup01.srv.local`
+* The clients:
+  * John Doe's desktop: `johndoe.clnt.local`
+  * Webserver 01: `web01.srv.local`
+  * Application server 01: `app01.srv.local`
+
+User and group
+--------------
+
+The repository server needs to have only one UNIX user for all the clients.
+Recommended user and group with additional settings:
+
+* User: `backup`
+* Group: `backup`
+* Shell: `/bin/bash` (or other capable to run the `borg serve` command)
+* Home: `/home/backup`
+
+Most clients shall initiate a backup from the root user to catch all
+users, groups and permissions (e.g. when backing up `/home`).
+
+Folders
+-------
+
+The following folder tree layout is suggested on the repository server:
+
+* User home directory, /home/backup
+* Repositories path (storage pool): /home/backup/repos
+* Clients restricted paths: `/home/backup/repos/<client fqdn>`
+  * johndoe.clnt.local: `/home/backup/repos/johndoe.clnt.local`
+  * web01.srv.local: `/home/backup/repos/web01.srv.local`
+  * app01.srv.local: `/home/backup/repos/app01.srv.local`
+
+Restrictions
+------------
+
+Borg is instructed to restrict clients into their own paths:
+``borg serve --restrict-path /home/backup/repos/<client fqdn>``
+
+There is only one ssh key per client allowed. Keys are added for ``johndoe.clnt.local``, ``web01.srv.local`` and
+``app01.srv.local``. But they will access the backup under only one UNIX user account as:
+``backup@backup01.srv.local``. Every key in ``$HOME/.ssh/authorized_keys`` has a
+forced command and restrictions applied as shown below:
+
+::
+
+  command="cd /home/backup/repos/<client fqdn>;
+           borg serve --restrict-path /home/backup/repos/<client fqdn>",
```

Comment:
well, wether --umask makes sense or not depends on what the default umask is and in some cases whether per-user groups is used or not. i introduce that option because i noticed that read permissions for "other" were given, which is undesirable for a backup storage (esp. if not encrypted). that's why it is now always used and defaults to a more safe "user-only" mask.

--verbose (on the server) should be used if the command on the client was invoked with it, otherwise you'll miss part of the expected output. I am not sure I can follow "it could expose info about the server".

Similar, it is for --lock-wait and --show-rc, they need to be given so that the server side behaves as expected.